lms/djangoapps/mobile_api/users/views.py

```diff
@@ -530,6 +533,133 @@ def my_user_info(request, api_version):
+                "course_name": "a",
+                "recently_active": true
+            },
+            {
+                "course_id": "course-v1:b+b+b",
+                "course_name": "b",
+                "recently_active": true
+            },
+            {
+                "course_id": "course-v1:c+c+c",
+                "course_name": "c",
+                "recently_active": false
+            },
+            ...
+        ]
+        ```
+    """
+
+    USER_ENROLLMENTS_LIMIT = 500
+
+    def get(self, request, *args, **kwargs) -> Response:
+        """
+        Gets user's enrollments status.
+        """
+        active_status_date = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=30)
+        username = kwargs.get('username')
+        course_ids_where_user_has_completions = self._get_course_ids_where_user_has_completions(
+            username,
+            active_status_date,
+        )
+        enrollments_status = self._build_enrollments_status_dict(
+            username,
+            active_status_date,
+            course_ids_where_user_has_completions
+        )
+        return Response(enrollments_status)
+
+    def _build_enrollments_status_dict(
+        self,
+        username: str,
+        active_status_date: datetime,
+        course_ids: List[str],
+    ) -> List[Dict[str, bool]]:
+        """
+        Builds list with dictionaries with user's enrolments statuses.
+        """
+        user = get_object_or_404(User, username=username)
+        user_enrollments = (
+            CourseEnrollment
+            .enrollments_for_user(user)
+            .select_related('course')
+            [:self.USER_ENROLLMENTS_LIMIT]
+        )
+        mobile_available = [
+            enrollment for enrollment in user_enrollments
+            if is_mobile_available_for_user(user, enrollment.course_overview)
+        ]
+        enrollments_status = []
+        for user_enrollment in mobile_available:
+            course_id = str(user_enrollment.course_overview.id)
+            enrollments_status.append(
+                {
+                    'course_id': course_id,
+                    'course_name': user_enrollment.course_overview.display_name,
+                    'recently_active': bool(
+                        course_id in course_ids
+                        or user_enrollment.created > active_status_date
+                    )
+                }
+            )
+        return enrollments_status
+
+    @staticmethod
+    def _get_course_ids_where_user_has_completions(
+        username: str,
+        active_status_date: datetime,
+    ) -> List[str]:
+        """
+        Gets course ids where user has completions.
+        """
+        context_keys = BlockCompletion.objects.filter(
+            user__username=username,
+            created__gte=active_status_date
+        ).values_list('context_key', flat=True).distinct()
+
+        return [str(context_key) for context_key in context_keys]
```

Comment:
First off, thanks for adding type annotations here 👍🏻 

Secondly, notice that you are converting these context keys to strings, and you are also converting the `user_enrollment.course_overview.id` to a string. You could make the whole system simpler and more type-safe by not doing that--just return ContextKey objects from this function, and then check whether the user enrollment's course id (which is also a ContextKey) belongs to that collection of objects.

Lastly, and this is just an optional nit, you could return a set of keys rather than a list, since your use of `distinct()` guarantees that there are no duplicates.

Reply:
I think the Set will also be a performance win since the loop on line 633 is using that list to determine if the course is active which is using the `in` operator which is faster for a set than a list.

Reply:
@kdmccormick @xitij2000 Hello again! I made the changes you wrote about above. Please take a look when you have time. Now we use Course `ContextKey` instead of id sting. Also I used Set with `context_keys`  instead of List.